Security/FeatureSecurity.php

```diff
@@ -54,4 +69,22 @@ public function isGranted(Feature $feature)
 
         return true;
     }
+
+    public function isGrantedForUser(Feature $feature, UserInterface $user): bool
+    {
+        $oldToken = $this->storage->getToken();
+
+        $this->storage->setToken(new UsernamePasswordToken(
+            $user,
+            null,
+            $this->providerKey,
+            $user->getRoles()
+        ));
+
+        $granted = $this->isGranted($feature);
```

Comment:
What about exception or errors?
If you don't catch anything in case of error the token storage will be in invalid state by holding the wrong user token

Reply:
As discussed in case of errors the token storage could be corrupted, but catching everything we will never see if the problem is happening nor when